augur/measurements.py

```diff
@@ -1,302 +1,23 @@
-    export_config.add_argument("--threshold", type=float,
-        help="A measurements value threshold to be displayed in the measurements panel.")
-    export_config.add_argument("--filters", nargs="+",
-        help="The columns that are to be used a filters for measurements. " +
-             "If not provided, all columns will be available as filters.")
-    export_config.add_argument("--group-by", type=str,
-        help="The default grouping column. If not provided, the first grouping will be used.")
-    export_config.add_argument("--measurements-display", type=str, choices=["raw", "mean"],
-        help="The default display of the measurements")
-
-    export_config.add_argument("--show-overall-mean", "--hide-overall-mean",
-        dest="show_overall_mean", action=HideAsFalseAction, nargs=0,
-        help="Show or hide the overall mean per group by default")
-    export_config.add_argument("--show-threshold", "--hide-threshold",
-        dest="show_threshold", action=HideAsFalseAction, nargs=0,
-        help="Show or hide the threshold by default. This will be ignored if no threshold is provided.")
-
-    export_optional = export.add_argument_group(
-        title="OPTIONAL SETTINGS"
-    )
-    export_optional.add_argument("--include-columns", nargs="+",
-        help="The columns to include from the collection TSV in the measurements JSON. " +
-             "Be sure to list columns that are used as groupings and/or filters. " +
-             "If no columns are provided, then all columns will be included by default.")
-    export_optional.add_argument("--minify-json", action="store_true",
-        help="Export JSON without indentation or line returns.")
-
-
-    concat = subparsers.add_parser("concat", help="Concatenate multiple measurements JSONs into a single JSON file")
-    concat_required = concat.add_argument_group(
-        title="REQUIRED"
-    )
-    concat_required.add_argument("--jsons", required=True, type=str, nargs="+", metavar="JSONs",
-        help="Measurement JSON files to concatenate.")
-    concat_required.add_argument("--output-json", required=True, metavar="JSON", type=str,
-        help="Output JSON file")
-
-    concat_optional = concat.add_argument_group(
-        title="OPTIONAL SETTINGS"
-    )
-    concat_optional.add_argument("--default-collection", type=str,
-        help="The key of the default collection to display. " +
-             "If not provided, the first collection of the first JSON file will be displayed")
-    concat_optional.add_argument("--minify-json", action="store_true",
-        help="Concatenate JSONs without indentation or line returns.")
+    add_default_command(parser)
 
+    subparsers = parser.add_subparsers(dest='subcommand')
+    add_subparsers(subparsers, COMMANDS)
 
 
 def run(args):
-    if args.subcommand == 'export':
-        return export_measurements(args)
-    if args.subcommand == "concat":
-        return concat_measurements(args)
+    return args.__command__.run(args)
```

Comment:
Huh, just realizing that since `utils.add_subparsers` includes https://github.com/nextstrain/augur/blob/cb7970b9a16f0c72eae324885d9a6315e4970eb0/augur/utils.py#L298

We no longer need to define a `run` function here, since the subcommands are directly run by the top level.

Reply:
After further consideration, I decided to drop `utils.add_subparsers`. It seemed like such a convoluted way to add subparsers...Just let each command do its own version of registering subparsers within `register_arguments`.